Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalMixRotatedLayer.cc

```diff
@@ -0,0 +1,597 @@
+#ifdef EDM_ML_DEBUG
+        edm::LogVerbatim("HGCalGeom") << "DDHGCalMixRotatedLayer: " << glog1.name() << " Tubs made of "
+                                      << materialTop_[ii] << " of dimensions " << cms::convert2mm(r1) << ", "
+                                      << cms::convert2mm(r2) << ", " << cms::convert2mm(hthickl) << ", "
+                                      << convertRadToDeg(phi1) << ", " << convertRadToDeg(phi2);
+#endif
+        dd4hep::Position tran(cshift.first, cshift.second, zpos);
+        glog.placeVolume(glog1, copy, tran);
+#ifdef EDM_ML_DEBUG
+        edm::LogVerbatim("HGCalGeom") << "DDHGCalMixRotatedLayer: Position " << glog1.name() << " number " << copy
+                                      << " in " << glog.name() << " at (0, 0, " << cms::convert2mm(zpos)
+                                      << ") with no rotation";
+#endif
+      }
+      ++copyNumberTop_[ii];
+      zpos += hthickl;
+    }
+    if (std::abs(thickTot - thick) > tol2_) {
+      if (thickTot > thick) {
+        edm::LogError("HGCalGeom") << "Thickness of the partition " << cms::convert2mm(thick) << " is smaller than "
+                                   << cms::convert2mm(thickTot)
+                                   << ": thickness of all its components in the top part **** ERROR ****";
+      } else {
+        edm::LogWarning("HGCalGeom") << "Thickness of the partition " << cms::convert2mm(thick)
+                                     << " does not match with " << cms::convert2mm(thickTot)
+                                     << " of the components in top part";
+      }
+    }
+
+    // Make the bottom part next
+    int layer = (copyM - firstLayer_);
+    static const double sqrt3 = std::sqrt(3.0);
+    int layercenter = (layerOrient_[layer] == HGCalTypes::CornerCenteredLambda)
+                          ? HGCalTypes::CornerCenterYp
+                          : ((layerOrient_[layer] == HGCalTypes::CornerCenteredY) ? HGCalTypes::CornerCenterYm
+                                                                                  : HGCalTypes::WaferCenter);
+    int layertype = (layerOrient_[layer] == HGCalTypes::WaferCenteredBack) ? 1 : 0;
+    int firstWafer = waferLayerStart_[layer];
+    int lastWafer = ((layer + 1 < static_cast<int>(waferLayerStart_.size())) ? waferLayerStart_[layer + 1]
+                                                                             : static_cast<int>(waferIndex_.size()));
+    double delx = 0.5 * (waferSize_ + waferSepar_);
+    double dely = 2.0 * delx / sqrt3;
+    double dy = 0.75 * dely;
+    const auto& xyoff = geomTools_.shiftXY(layercenter, (waferSize_ + waferSepar_));
+#ifdef EDM_ML_DEBUG
+    int ium(0), ivm(0), kount(0);
+    std::vector<int> ntype(3, 0);
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalMixRotatedLayer: " << glog.name() << "  r " << cms::convert2mm(delx)
+                                  << " R " << cms::convert2mm(dely) << " dy " << cms::convert2mm(dy) << " Shift "
```

Comment:
Are you restoring the "r" and "R" names for consistency with the previous version?

Reply:
Yes